content/docs/projects/deals/index.md

```diff
@@ -0,0 +1,295 @@
+
+- Item Buying Deals
+- Item Selling Deals
+
+## 2.2 Validations
+
+Frontend and backend validations are present to validate values upon entering as well as as upon saving. 
+
+### 2.2.1 Item Deals Validation
+
+```python
+
+
+class Item(WebsiteGenerator):
+    def validate(self):
+        self.validate_deals()
+
+     def validate_deals(self):
+        for deal in self.buying_deals:
+            deal.validate(self)
+
+        for deal in self.selling_deals:
+            deal.validate(self)
+```
+
+```python 
+class ItemSellingDeal(Document):
+    def validate(self, item):
+        # validates qty (if qty greater than 0)
+        # validates uom (if UOM is maintained in the parent item)
+        # validates the item (if there is an item entered in the field)
+        # validates the free item uom (if the free item UOM is maintained in the free item's profile)
+```
+
+```python 
+class ItemBuyingDeal(Document):
+    def validate(self, item):
+        # validates the supplier (if supplier is maintained in the item profile)
+        # validates qty (if qty greater than 0)
+        # validates uom (if UOM is maintained in the parent item)
+        # validates the item (if there is an item entered in the field)
+        # validates the free item uom (if the free item UOM is maintained in the free item's profile)
+```
+
+On Item.js the quantities are validated upon enter. 
+```javascript
+frappe.ui.form.on("Item Selling Deal", {
+        free_qty: async function (frm, cdt, cdn) {
+            await validate_for_negative_qty(frm, cdt, cdn, "free_qty"); 
+        },
+        min_qty: async function (frm, cdt, cdn) {
+            await validate_for_negative_qty(frm, cdt, cdn, "min_qty");
+        },
+});
+```
+
+
+## 2.3 Pricing Rule Controller
+
+The pricing rule controller operates similarly to the discount controller. It utilizes the same code, workflow, and processes. For further details on the pricing rule controller, please refer to section 2.3, "Pricing Rule Controller," in the discount documentation: link to documentation: <a href="https://bizkit-tech.github.io/docs/projects/discounts/#23-pricing-rule-controller"> https://bizkit-tech.github.io/docs/projects/discounts/#23-pricing-rule-controller</a>.
+
+# 3 Application To Transactions
+
+Item deals are appear when you add an item into the items table of a certain transaction. The intial application of a deal is when an Item Code is entered and since it always defaults to one, it then checks for a deal that applicable to the item. This happens through an API call. 
+
+There is only one touchpoint on a deal application. This is when an 
+
+1. Item Code in the Items table is changed
+
+   Inside the `item_code` form event of `erpnext.TransactionController`, the `apply_item_deals` function of the `erpnext.TransactionController` class is called after the item details are fetched from the server call.
+   ```js
+   apply_item_deals: function (item) {

```

Comment:
Is it possible na instead of pasting the entire function is hatiin mo na lang siya into snippets tapos yung corresponding comment/explanation below the snippet